### ISSUE TITLE
feature request. animation with slugtrails

### ISSUE THREAD

**Issue by pssguy** [ropensci/plotly#957]
By this, I mean as the marker in an animation advances it leaves a line behind traversing all the previous points

**Comment by cpsievert** [comment]
I agree, this would be useful, and possibly not that hard implement.

Any ideas for how to specify? One option is the [gganimate approach](https://github.com/dgrtwo/gganimate/#customization), but I hate to keep overloading ggplot2 aesthetics with non-standard semantics...

**Comment by cpsievert** [comment]
After some more thought, I'm not sure there is a fool-proof way to approach this with the existing plotly.js frame API. 

For something fairly simple, like building a path overtime, you could replicate rows in your data in a clever way, like this:

```r
accumulate_by <- function(dat, var) {
  var <- lazyeval::f_eval(var, dat)
  lvls <- plotly:::getLevels(var)
  dats <- lapply(seq_along(lvls), function(x) {
    cbind(dat[var %in% lvls[seq(1, x)], ], frame = lvls[[x]])
  })
  dplyr::bind_rows(dats)
}

library(plotly)
d <- txhousing %>%
  filter(date < 2002) %>%
  accumulate_by(~date)

p <- ggplot(d, aes(date, median)) + 
  geom_line(aes(group = city, ids = city, frame = frame))
ggplotly(p)
```

![ani-bad](https://cloud.githubusercontent.com/assets/1365941/25255329/915da00e-25ef-11e7-84f6-de817c40abdb.gif)


@rreusser, it looks like [the JSON this creates should ensure than none of the lines shift, but they do](http://rpubs.com/cpsievert/269603), any ideas? Also, I don't suppose you're planning on working on this (adding cumulative semantics to the API) anytime soon?

**Comment by cpsievert** [comment]
@rreusser the shifting lines must have to do with the fact that the previous plot is one trace with multiple `null` values to separate the lines. The problem goes away when changing `group` to `color` (effectively creating multiple traces).

![ani-good](https://cloud.githubusercontent.com/assets/1365941/25255468/643810ae-25f0-11e7-8b06-945524e82456.gif)

**Comment by cpsievert** [comment]
Thanks @rreusser, an example would be very helpful! Also, this doesn't seem to fix the problem:

```r
style(ggplotly(p), list(line = list(simplify = FALSE)))
```

**Comment by pssguy** [comment]
Thanks guys. Adapted for use in [blog post](https://www.mytinyshinys.com/post/plotlyFunctions/)

![ptsanimation](https://cloud.githubusercontent.com/assets/2775918/25289763/d37c7308-267f-11e7-8dc9-3a9b75cd9ed0.png)

**Comment by cpsievert** [comment]
Awesome post! Looks like the first image is missing?

**Comment by rreusser** [comment]
Here's an example that uses a filter transform to slice a dataset *without actually precomputing each independent slice*: https://codepen.io/rsreusser/pen/ZKOGdz?editors=0010

**Comment by ismirsehregal** [comment]
I'm currently trying to create cumulative animations without accumulating the data for each frame in R ([Here](https://stackoverflow.com/questions/49871541/is-there-any-other-way-to-implement-plotly-r-s-cumulative-animation/55922974#55922974) is the SO question). Depending on the use case the `accumulate_by()` approach builds up too much data.

Is there a way to create custom steps for the `animation_slider()` to e.g. implement a filter transform as mentioned above by @rreusser or as done [here](https://community.plot.ly/t/animation-produces-extremely-large-files/14812/8)? - If not, that would be a really useful feature.

**Comment by ismirsehregal** [comment]
For those interested, here is an example on how to use filter transform and a custom range slider in R, however, still no animation:

```
DF <- data.frame(x = rep(1:10, 2), y = runif(20), group = rep(LETTERS[1:2], each = 10))

steps <- list()

for (i in seq_along(unique(DF$x))) {
  steps[[i]] <- list(
    args = list('transforms[0].value', i),
    label = i,
    method = "restyle",
    value = i
  )
}

p_filter_slider <- plot_ly(
  DF,
  x = ~ x,
  y = ~ y,
  color = ~ group,
  type = "scatter",
  mode = "lines+markers",
  transforms = list(
    list(
      type = 'filter',
      target = 'x',
      operation = '<=',
      value = ~ x
    )
  )
) %>% layout(title = "Custom filter slider",
             xaxis = list(range = c(0.5, length(unique(DF$x))+0.5)),
             sliders = list(
               list(
                 active = 0, 
                 currentvalue = list(prefix = "X-max: "), 
                 pad = list(t = 20), 
                 steps = steps))
             )

p_filter_slider
```
![screen](https://user-images.githubusercontent.com/36849480/76218401-7b8e1b00-6214-11ea-8fe2-46e87eab9036.gif)
